src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1116,7 +1116,7 @@ bool isRefEscape
             // check receiver if ref-like
             if (receiverOpt?.Type?.IsByRefLikeType == true)
             {
-                return GetValEscape(receiverOpt, scopeOfTheContainingExpression);
+                escapeScope = Math.Max(escapeScope, GetValEscape(receiverOpt, scopeOfTheContainingExpression));
```

Comment:
>escapeScope [](start = 16, length = 11)

I'll stop by. I didn't fully understand the issue.

Also, does any of the specs need to be updated according to the fixes you are making to 15.8?
(maybe [https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.2/span-safety.md](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.2/span-safety.md) or [https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.3/ref-local-reassignment.md](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.3/ref-local-reassignment.md) ?)

Reply:
@jcouv Check the `Method Invocation` section. It says "the smallest of the following scopes" from all arguments including the receiver. The current code ignores the arguments escape scope when receiver is ref-like. I believe the spec is up to date.

Reply:
cc @gafter